### PR TITLE
CDAP-11633 upgrade tool prevent upgrade if master is running

### PR DESCRIPTION
**Pull request by gsps1**

### CODE REVIEW

**Comment by gsps1** [comment]
build running : https://builds.cask.co/browse/CDAP-RUT1066-1

File: cdap-master/src/main/java/co/cask/cdap/data/tools/UpgradeTool.java [hunk 1]
```diff
@@ -361,6 +374,31 @@ private void doMain(String[] args) throws Exception {
     }
   }
 
+  /**
+   * checks for appfabric service path on zookeeper, if they exist, CDAP master is still running, so throw
```
Comment (anew) [inline cdap-master/src/main/java/co/cask/cdap/data/tools/UpgradeTool.java]: Checks (upper case)

File: cdap-master/src/main/java/co/cask/cdap/data/tools/UpgradeTool.java [hunk 2]
```diff
@@ -361,6 +374,31 @@ private void doMain(String[] args) throws Exception {
...
+   * exception message with information on where its running.
+   * If none of the master service (in HA setup) is running, this method wouldn't throw any exception
```
Comment (anew) [inline cdap-master/src/main/java/co/cask/cdap/data/tools/UpgradeTool.java]: this line is unnecessary.

File: cdap-master/src/main/java/co/cask/cdap/data/tools/UpgradeTool.java [hunk 3]
```diff
@@ -361,6 +374,31 @@ private void doMain(String[] args) throws Exception {
...
+   * @throws Exception
```
Comment (anew) [inline cdap-master/src/main/java/co/cask/cdap/data/tools/UpgradeTool.java]: if at least one master is running

File: cdap-master/src/main/java/co/cask/cdap/data/tools/UpgradeTool.java [hunk 4]
```diff
@@ -361,6 +374,31 @@ private void doMain(String[] args) throws Exception {
...
+        NodeData nodeData = zkClientService.getData(String.format("%s/%s", appFabricPath, runId)).get();
+        Discoverable discoverable = GSON.fromJson(Bytes.toString(nodeData.getData()), Discoverable.class);
+        runningNodes.add(discoverable.getSocketAddress().getHostName());
+      }
+      String exceptionMessage =
+        String.format("CDAP Master is still running on %s, please stop it before running upgrade.",
+                      com.google.common.base.Joiner.on(",").join(runningNodes));
+      throw new Exception(exceptionMessage);
+    }
+  }
```
Comment (anew) [inline cdap-master/src/main/java/co/cask/cdap/data/tools/UpgradeTool.java]: As a future improvement, the upgrade tool register as a master to become the leader and prevent other masters from starting.

**Review by anew — COMMENTED** [review]
some minor comments, otherwise looks good

**Comment by gsps1** [comment]
build : https://builds.cask.co/browse/CDAP-RUT1066-1/log

**Review by anew — APPROVED** [review]
LGTM

**Comment by sreevatsanraman** [comment]
Build passed. Merging.

**Pull request merged**